automated install/basic-install.sh

```diff
@@ -88,16 +88,27 @@ if command -v apt-get &> /dev/null; then
   DNSMASQ_USER="dnsmasq"
 
   test_dpkg_lock() {
-    i=0
-    while fuser /var/lib/dpkg/lock >/dev/null 2>&1 ; do
-      sleep 0.5
-      ((i=i+1))
+    # Account for cloud init type installs that autorun apt on boot
+    local maxtries
+    local retVal
+    maxtries=75
+
+    for i in $(seq 1 $maxtries);
+    do
+      "${UPDATE_PKG_CACHE}"
+      retVal=$?
+      # if we find lock condition
+      if [[ $retVal -eq 100 ]]; then
+        sleep 5
+      elif [[ $retVal -eq 0 ]]; then
+        return 0
+      fi
     done
-    # Always return success, since we only return if there is no
-    # lock (anymore)
-    return 0
+    # Counter timed out, and we didn't return already, error
+    return 1
```

Comment:
I see we wait up to 6 minutes - is this sufficient in case that a huge bunch of security updates is installed on low-performance (possibly including more than one kernel and some `dkms`) are downloaded and compiled/installed? 

From the `man` page:

> apt-get returns zero on normal operation, decimal 100 on error.

Maybe we can still check for the lock and at least tell the user *why* he has to wait?